### PR TITLE
refactor: modify functions with return values,reduce redundant code

### PR DESCRIPTION
**Pull request by songshuone**
Signed-off-by:  Wu Peng    1399094187@qq.com
###
### Ⅰ. Describe what this PR did

modify functions with return values,reduce redundant code

### CODE REVIEW

**Comment by fengzixu** [comment]
Please fulfil the information in PR Description according to the outline and revise your commit log. If you feel confused, please read guide doc: https://github.com/dragonflyoss/Dragonfly/blob/master/CONTRIBUTING.md

File: apis/types/peer_create_request.go [hunk 1]
```diff
@@ -85,64 +85,48 @@ func (m *PeerCreateRequest) Validate(formats strfmt.Registry) error {
 	return nil
 }
 
-func (m *PeerCreateRequest) validateIP(formats strfmt.Registry) error {
```
Comment (fengzixu) [inline apis/types/peer_create_request.go]: Please fulfil the information in PR Description according to the outline and revise your commit log. If you feel confused, please read guide doc: https://github.com/dragonflyoss/Dragonfly/blob/master/CONTRIBUTING.md

Comment (fengzixu) [reply apis/types/peer_create_request.go]: A better choice: describe clearly why did you modify those logics. It is beneficial for reviewers.

Comment (songshuone) [reply apis/types/peer_create_request.go]: Okay, let me read the guide.

Comment (fengzixu) [reply apis/types/peer_create_request.go]: Before you submit your code, you can run some commands, such as `make unit-test or make build and so on` which can help you test your code in your local developing environment.

Comment (songshuone) [reply apis/types/peer_create_request.go]: ok

**Comment by songshuone** [comment]
> ping @songshuone
> Conflict happens after merging a previous commit.
> Please rebase the branch against master and push it back again. Thanks a lot.
ok

**Comment by fengzixu** [comment]
Hey @songshuone. You need to notice the comment written by @pouchrobot .

1. Please sign off in each of your commits.
2. Please resolve the confilct.

PS: Before you develop some features base on your repo, you should rebase the newest code about Dragonfly in the master branch. All developers develop features base on the master branch.

**Comment by fengzixu** [comment]
Some personal advice for you:

1. If you were developing the function and not ready for the code review, you should add `WIP` tag in the title of the PR.
2. After your PR is ready and all checks are successful, you can request the review. That is an effective approach.

File: apis/types/error.go [hunk 2]
```diff
@@ -20,24 +20,24 @@ type Error struct {
 }
 
 // Validate validates this error
```
Comment (fengzixu) [inline apis/types/error.go]: Your commit message has exceeded one line，please use one statement to describe you PR. And you can write more contents in commit content. Besides, your commit message looks like no sense：

```
refactor: api/types/*.go code review, modify functions with return values
```
1. What is code review ?

File: apis/types/error.go [hunk 3]
```diff
@@ -20,24 +20,24 @@ type Error struct {
...
-func (m *Error) Validate(formats strfmt.Registry) error {
-	return nil
+func (m *Error) Validate(formats strfmt.Registry) (err error) {
```
Comment (fengzixu) [inline apis/types/error.go]: Why did you revise there? Using the anonymous return value in the signature of the function is ok, especially we didn't add significant logics.

File: apis/types/error.go [hunk 4]
```diff
@@ -20,24 +20,24 @@ type Error struct {
...
+	return err
 }
 
 // MarshalBinary interface implementation
-func (m *Error) MarshalBinary() ([]byte, error) {
+func (m *Error) MarshalBinary() (b []byte, error) {
```
Comment (fengzixu) [inline apis/types/error.go]: It looks like so wired. One return parameter is anonymous and another one has a name.

File: apis/types/error.go [hunk 5]
```diff
@@ -20,24 +20,24 @@ type Error struct {
...
 }
 
 // UnmarshalBinary interface implementation
-func (m *Error) UnmarshalBinary(b []byte) error {
+func (m *Error) UnmarshalBinary(b []byte) (err  error) {
 	var res Error
-	if err := swag.ReadJSON(b, &res); err != nil {
-		return err
+	if err = swag.ReadJSON(b, &res); err != nil {
+		return
```
Comment (fengzixu) [inline apis/types/error.go]: Why did you revise there? You definite a variable whose type is the error, but you didn't return error info?

**Pull request closed without merging**